File: src/emitter/mermaid/collect-import-sources.ts
import { NODE_KIND } from "../../visual-graph/node-kind.js";
import type { VisualNode } from "../../visual-graph/visual-node.js";

export function collectImportSources(
  nodeMap: ReadonlyMap<string, VisualNode>,
): Set<string> {
  const ids = new Set<string>();
  for (const n of nodeMap.values()) {
    if (
      n.kind === NODE_KIND.LegacyModuleSource ||
      n.kind === NODE_KIND.LegacyImportIntermediate
    ) {
      ids.add(n.id);
    }
  }
  return ids;
}

File: crates/unsnarl-emitter-mermaid/src/collect_import_sources.rs
//! Selects the set of ids whose outgoing edges should be rendered
//! in the import-edge bucket.

use std::collections::{HashMap, HashSet};

use unsnarl_visual_graph::node_kind::NodeKind;
use unsnarl_visual_graph::visual_node::VisualNode;

pub fn collect_import_sources(node_map: &HashMap<String, &VisualNode>) -> HashSet<String> {
    let mut out = HashSet::new();
    for (id, n) in node_map {
        match n.kind() {
            NodeKind::SyntheticModuleSource | NodeKind::SyntheticImportIntermediate => {
                out.insert(id.clone());
            }
            _ => {}
        }
    }
    out
}

#[cfg(test)]
#[path = "collect_import_sources_test.rs"]
mod collect_import_sources_test;
